src/main/java/cn/nukkit/block/BlockLog.java

```diff
@@ -45,21 +62,30 @@ public int getBurnAbility() {
 
     @Override
     public boolean place(Item item, Block block, Block target, BlockFace face, Vector3f clickPos, Player player) {
-        short[] faces = new short[]{
-                0,
-                0,
-                0b1000,
-                0b1000,
-                0b0100,
-                0b0100
-        };
-
-        this.setDamage(((this.getDamage() & 0x03) | faces[face.getIndex()]));
+        this.setDamage(((this.getDamage() & 0x03) | FACES[face.getIndex()]));
         this.getLevel().setBlock(block, this, true, true);
 
         return true;
     }
 
+    @Override
+    public boolean canBeActivated() {
+        return true;
+    }
+
+    @Override
+    public boolean onActivate(Item item, Player player) {
+        if (!item.isAxe()) {
+            return false;
+        }
+
+        int log2Damage = this instanceof BlockLog2 ? 4 : 0;
+        int damage = ( this.getDamage() & -0b100 ) ^ this.getDamage();
+        Block strippedBlock = Block.get(STRIPPED_IDS[damage + log2Damage], ( this.getDamage() >> 2 ) );
+        this.getLevel().setBlock(this.asVector3i(), strippedBlock, true, true);
+        return true;
+    }
+
```

Comment:
Almost perfect, it's just not damaging the axe. I'm going to fix in a few minutes.